app/src/main/java/com/courseproject/tindar/ChatActivity.java

```diff
@@ -1,35 +1,178 @@
+
+        this.chatRecyclerView = findViewById(R.id.chat_recycler_view); // matches recycler to layout
+        this.chatRecyclerView.setAdapter(adapter);
+        this.chatRecyclerView.setLayoutManager(linearLayoutManager);
+    }
+
+    // TODO: this should be implemented properly
+    // and call a different class for the messages. probably conversation.
+    /**
+     * Loads or adds to the list of messages to be displayed,
+     * which is stored in the instance variable loadedMessages.
+     * Called by OnCreate. Should also be called whenever the user wants to load more messages.
+     *
+     * Currently implemented with hardcoded tests; will be reimplemented with the database.
+     *
+     * @return void
+     */
+    private void loadMessages(){
+        TindarMessage testMessage1 = new TindarMessage("first message sent", new Timestamp(2023, 02, 25, 18, 0, 0, 0),"user1", "user2");
+        TindarMessage testMessage2 = new TindarMessage("second message sent", new Timestamp(2023, 03, 25, 18, 0, 0, 0),"user1", "user2");
+        TindarMessage testMessage3 = new TindarMessage("first message received", new Timestamp(2023, 04, 25, 18, 0, 0, 0),"user2", "user1");
+        TindarMessage testMessage4 = new TindarMessage("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam maximus erat nisl, a scelerisque leo euismod nec. Ut dapibus auctor augue, quis tempor tellus tincidunt vel. Fusce ut odio mauris. Nam nec finibus enim. Duis et nisi tristique, luctus leo id, facilisis dolor. Phasellus ac auctor odio, non mollis magna. Suspendisse tortor ipsum, consectetur vitae metus et, accumsan luctus erat. Sed gravida, ipsum vel mattis maximus, orci arcu convallis nunc, sed sagittis metus felis ac sapien. Integer non pellentesque massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed et dignissim erat. Integer at nibh ac mi ultricies pulvinar. Morbi nec arcu nisi. Duis eu ligula auctor, dictum tortor a, condimentum velit. Suspendisse potenti. Sed fermentum lobortis blandit.", new Timestamp(2023, 05, 25, 18, 0, 0, 0),"user1", "user2");
+        TindarMessage testMessage5 = new TindarMessage("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam maximus erat nisl, a scelerisque leo euismod nec. Ut dapibus auctor augue, quis tempor tellus tincidunt vel. Fusce ut odio mauris. Nam nec finibus enim. Duis et nisi tristique, luctus leo id, facilisis dolor. Phasellus ac auctor odio, non mollis magna. Suspendisse tortor ipsum, consectetur vitae metus et, accumsan luctus erat. Sed gravida, ipsum vel mattis maximus, orci arcu convallis nunc, sed sagittis metus felis ac sapien. Integer non pellentesque massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed et dignissim erat. Integer at nibh ac mi ultricies pulvinar. Morbi nec arcu nisi. Duis eu ligula auctor, dictum tortor a, condimentum velit. Suspendisse potenti. Sed fermentum lobortis blandit.", new Timestamp(2023, 05, 25, 18, 0, 0, 0),"user2", "user1");
```

Comment:
will these huge test chat messages remain in the final implementation?

Reply:
absolutely not lmao

Reply:
at least not here. I might move them to unit tests.